src/main/java/sirius/biz/jobs/batch/file/ArchiveImportJob.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.jobs.batch.file;
+
+import sirius.biz.process.ProcessContext;
+import sirius.biz.process.logs.ProcessLog;
+import sirius.biz.storage.layer1.FileHandle;
+import sirius.biz.storage.layer3.FileParameter;
+import sirius.kernel.commons.Files;
+import sirius.kernel.health.Exceptions;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * Provides an import job which allows to import data from multiple archived files in a specific order.
+ */
+public abstract class ArchiveImportJob extends FileImportJob {
+
+    /**
+     * Contains the file name and input stream of all archive entries whose extensions can be handled.
+     */
+    private final Map<String, InputStream> entries = new HashMap<>();
+
+    /**
+     * Creates a new job for the given process context.
+     *
+     * @param fileParameter the parameter which is used to derive the import file from
+     * @param process       the process context in which the job is executed
+     */
+    protected ArchiveImportJob(FileParameter fileParameter, ProcessContext process) {
+        super(fileParameter, process);
+    }
+
+    @Override
+    protected void executeForSingleFile(FileHandle fileHandle) throws Exception {
+        entries.put(fileHandle.getFile().getName(), fileHandle.getInputStream());
```

Comment:
Shouldn‘t we rather throw/abort here - we acutally expect an archive with multiple files answay

Reply:
This might generate quite a few changes in the job structure further down, but will simplify the logic here. I'll change it and see how much refactoring is actually needed.